pandas/core/nanops.py

```diff
@@ -326,14 +329,19 @@ def nanmean(values, axis=None, skipna=True):
 @bottleneck_switch()
 def nanmedian(values, axis=None, skipna=True):
 
+    if not skipna and isnull(np.sum(values)):
+        return np.median(values, axis=axis)
```

Comment:
Is this special case needed? As `np.nanmedian` is almost as fast as `np.median`, also when there are no NaNs.